connaisseur/alert.py

```diff
@@ -58,48 +57,49 @@ class AlertReceiverAuthentication:
     authentication_scheme: str = None
 
     class AlertReceiverAuthenticationInterface:
-        def __init__(self, alert_receiver_config: dict, authentication_key: str):
-            self.authentication_config = alert_receiver_config.get(authentication_key)
-
-            if self.authentication_config is None:
-                raise ConfigurationError(
-                    f"No authentication configuration found ({authentication_key})."
+        def __init__(self, alert_receiver_config: dict, authentication_config_key: str):
+            if authentication_config_key is not None:
+                self.authentication_config = alert_receiver_config.get(
+                    authentication_config_key
                 )
 
-            self.authentication_scheme = self.authentication_config.get(
-                "authentication_scheme", self.authentication_scheme
-            )
-            self._validate_authentication_scheme()
+                if self.authentication_config is None:
+                    raise ConfigurationError(
+                        "No authentication configuration found for dictionary key:"
+                        f"{authentication_config_key}."
+                    )
```

Comment:
Here, you can also `try: self.authentication_config = alert_receiver_config[authentication_config_key]`, and then raise the error in the `except:` block.

Reply:
Fixed also other places where EAFP can be applied.